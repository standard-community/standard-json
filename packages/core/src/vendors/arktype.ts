import type { Type } from "arktype";
import type { JSONSchema7 } from "json-schema";
import type { ToJsonSchemaFn } from "./index.js";

export const getToJsonSchemaFn =
  async (): Promise<ToJsonSchemaFn> => (schema) =>
    (schema as Type).toJsonSchema() as JSONSchema7;
