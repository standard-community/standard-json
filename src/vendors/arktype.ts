import type { Type } from "arktype";
import type { JSONSchema7 } from "json-schema";
import type { ToJsonSchemaFn } from "./utils.js";

export default async function getToJsonSchemaFn(): Promise<ToJsonSchemaFn> {
  return (schema, options) =>
    (schema as Type).toJsonSchema(options) as JSONSchema7;
}
