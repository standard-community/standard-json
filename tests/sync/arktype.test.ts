import { type Type, type } from "arktype";
import { describe, expect, it } from "vitest";

import { loadVendor, toJsonSchema } from "~/index.js";
import type { ToJsonSchemaFn } from "~/vendors/utils.js";

describe("sync check", () => {
  it("arktype", () => {
    // Loading the vendor
    loadVendor("arktype", ((schema, options) =>
      (schema as Type).toJsonSchema(options)) as ToJsonSchemaFn,
    );

    const schema = type({
      myString: "string",
      myUnion: "number | boolean",
    }).describe("My neat object schema");

    const jsonSchema = toJsonSchema(schema);
    expect(jsonSchema).toMatchSnapshot();
  });
});
