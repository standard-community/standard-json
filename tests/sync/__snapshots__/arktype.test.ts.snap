// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`sync check > arktype 1`] = `
{
  "$schema": "https://json-schema.org/draft/2020-12/schema",
  "description": "My neat object schema",
  "properties": {
    "myString": {
      "type": "string",
    },
    "myUnion": {
      "anyOf": [
        {
          "type": "number",
        },
        {
          "type": "boolean",
        },
      ],
    },
  },
  "required": [
    "myString",
    "myUnion",
  ],
  "type": "object",
}
`;
